src/System.Private.ServiceModel/tests/Scenarios/Binding/Custom/AsyncResult.cs

```diff
@@ -0,0 +1,236 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+/// <summary>
+/// A generic base class for IAsyncResult implementations
+/// that wraps a ManualResetEvent.
+/// </summary>
+abstract class AsyncResult : IAsyncResult
```

Comment:
Could you briefly describe where these files came from?  Are they copied from a sample?  Product code?  I'm asking about AsyncResult.cs, ChainedAsyncResult.cs, and TimeoutHelper.cs mostly.